modules/configuring-your-ide.adoc

```diff
@@ -0,0 +1,199 @@
+
+link:https://marketplace.visualstudio.com/items?itemName=ms-ossdata.vscode-postgresql[See in Visual Studio Marketplace]
+
+SQLTools::
+SQLTools simplifies database connections and enhances data management and query execution, supporting a variety of commonly used databases.
+
+link:https://marketplace.visualstudio.com/items?itemName=mtxr.sqltools[See in Visual Studio Marketplace]
+
+Database Client::
+This extension serves as a database manager for MySQL/MariaDB, PostgreSQL, SQLite, Redis, and ElasticSearch. It provides a unified interface for managing diverse databases within the VS Code Server environment.
+
+link:https://marketplace.visualstudio.com/items?itemName=cweijan.vscode-database-client2[See in Visual Studio Marketplace]
+
+=== Incorporating DBMS clients using a custom image
+
+Custom notebook images are useful when working with containerized environments, especially when you need specific libraries, OS packages, or applications that are not available in base images. Follow these steps to create a custom notebook image with Database Management System (DBMS) clients for MongoDB, MSSQL, and PostgreSQL using a Dockerfile.
+
+.Prerequisites
+
+* You have logged in to {productname-long}.
+ifndef::upstream[]
+* If you use specialized {productname-short} groups, you are part of the user group or admin group (for example, {oai-user-group} or {oai-admin-group}) in OpenShift.
+endif::[]
+ifdef::upstream[]
+* If you use specialized {productname-short} groups, you are part of the user group or admin group (for example, {odh-user-group} or {odh-admin-group}) in OpenShift.
+endif::[]
+* You have created a data science project that has a Code Server workbench.
+* You have installed Docker on your system.
+
+.Procedure
+
+. Clone the Open Data Hub *notebooks* repository:
++
+----
+git clone git@github.com:opendatahub-io/notebooks.git
+----
+
+. From your local version of the repository, go to the `codeserver` folder and create a new folder for your custom image. For example, `ubi9-python-3.9-db-clients`. 
+. Inside this folder, create a Dockerfile with the following contents:
++
+----
+# The base image is auto-assigned by the make recipe on the next step. In this case, this is your Code Server notebook.
+
+ARG BASE_IMAGE
+FROM ${BASE_IMAGE}
+
+WORKDIR /opt/app-root/bin
+
+# Install OS packages as root
+USER root
+
+# Install necessary OS packages
+RUN dnf install -y unixODBC postgresql
+
+# Install MongoDB Client
+COPY mongodb-org-6.0.repo-x86_64 /etc/yum.repos.d/mongodb-org-6.0.repo
+RUN dnf install -y mongocli
+
+# Install MSSQL Client
+COPY mssql-2022.repo-x86_64 /etc/yum.repos.d/mssql-2022.repo
+RUN ACCEPT_EULA=Y dnf install -y mssql-tools18 unixODBC-devel
+ENV PATH="$PATH:/opt/mssql-tools18/bin"
+
+# Switch back to default user
+USER 1001
+
+WORKDIR /opt/app-root/src
+----
+
+. Create two RPM files, `mongodb-org-6.0.repo-x86_64` and `mssql-2022.repo-x86_64`, with the following contents, in the folder you created for your custom image.
```

Comment:
These aren't `RPM` files. These are just text files with the configuration for yum/dnf for particular RPM repository.

Reply:
Good spot! Will correct this one.